src/arch/InputHandler/InputHandler_SDL.cpp

```diff
@@ -0,0 +1,974 @@
+        case JOY_BUTTON_5: btn = SDL_GAMEPAD_BUTTON_START;
+            break;
+        case JOY_BUTTON_6: btn = SDL_GAMEPAD_BUTTON_BACK;
+            break;
+        case JOY_BUTTON_7: btn = SDL_GAMEPAD_BUTTON_LEFT_STICK;
+            break;
+        case JOY_BUTTON_8: btn = SDL_GAMEPAD_BUTTON_RIGHT_STICK;
+            break;
+        case JOY_BUTTON_9: btn = SDL_GAMEPAD_BUTTON_LEFT_SHOULDER;
+            break;
+        case JOY_BUTTON_10: btn = SDL_GAMEPAD_BUTTON_RIGHT_SHOULDER;
+            break;
+        case JOY_BUTTON_11: axisBtn = SDL_GAMEPAD_AXIS_LEFT_TRIGGER;
+            break;
+        case JOY_BUTTON_12: axisBtn = SDL_GAMEPAD_AXIS_RIGHT_TRIGGER;
+            break;
+        case JOY_BUTTON_13: btn = SDL_GAMEPAD_BUTTON_GUIDE;
+            break;
+        case JOY_BUTTON_14: btn = SDL_GAMEPAD_BUTTON_MISC1;
+            break;
+        case JOY_BUTTON_15: btn = SDL_GAMEPAD_BUTTON_RIGHT_PADDLE1;
+            break;
+        case JOY_BUTTON_16: btn = SDL_GAMEPAD_BUTTON_LEFT_PADDLE1;
+            break;
+        case JOY_BUTTON_17: btn = SDL_GAMEPAD_BUTTON_RIGHT_PADDLE2;
+            break;
+        case JOY_BUTTON_18: btn = SDL_GAMEPAD_BUTTON_LEFT_PADDLE2;
+            break;
+        case JOY_BUTTON_19: btn = SDL_GAMEPAD_BUTTON_MISC2;
+            break;
+        case JOY_BUTTON_20: btn = SDL_GAMEPAD_BUTTON_MISC3;
+            break;
+        case JOY_BUTTON_21: btn = SDL_GAMEPAD_BUTTON_MISC4;
+            break;
+        case JOY_BUTTON_22: btn = SDL_GAMEPAD_BUTTON_MISC5;
+            break;
+        case JOY_BUTTON_23: btn = SDL_GAMEPAD_BUTTON_MISC6;
+            break;
+        case JOY_BUTTON_24: btn = SDL_GAMEPAD_BUTTON_TOUCHPAD;
+            break;
+        case JOY_HAT_UP: btn = SDL_GAMEPAD_BUTTON_DPAD_UP;
+            break;
+        case JOY_HAT_DOWN: btn = SDL_GAMEPAD_BUTTON_DPAD_DOWN;
+            break;
+        case JOY_HAT_LEFT: btn = SDL_GAMEPAD_BUTTON_DPAD_LEFT;
+            break;
+        case JOY_HAT_RIGHT: btn = SDL_GAMEPAD_BUTTON_DPAD_RIGHT;
+            break;
+        case JOY_LEFT:
+            axisBtn = SDL_GAMEPAD_AXIS_LEFTX;
+            axisPositive = false;
+            break;
+        case JOY_RIGHT: axisBtn = SDL_GAMEPAD_AXIS_LEFTX;
+            break;
+        case JOY_UP:
+            axisBtn = SDL_GAMEPAD_AXIS_LEFTY;
+            axisPositive = false;
+            break;
+        case JOY_DOWN: axisBtn = SDL_GAMEPAD_AXIS_LEFTY;
+            break;
+        case JOY_LEFT_2:
+            axisBtn = SDL_GAMEPAD_AXIS_RIGHTX;
+            axisPositive = false;
+            break;
+        case JOY_RIGHT_2: axisBtn = SDL_GAMEPAD_AXIS_RIGHTX;
+            break;
+        case JOY_UP_2:
+            axisBtn = SDL_GAMEPAD_AXIS_RIGHTY;
+            axisPositive = false;
+            break;
+        case JOY_DOWN_2: axisBtn = SDL_GAMEPAD_AXIS_RIGHTY;
+            break;
+        default: break;
+        }
+
+        // Handle face buttons
+        switch (SDL_GetGamepadButtonLabel(pGamepad, btn))
```

Comment:
It might be better if the gamepad / joystick type are exposed to themes, and have the themes do this work. That way the button labels can be localized (or use button glyphs instead of text :smile:)

I'm happy to rip this out and look into that for in a separate ticket